src/puffin/src/file_format/writer/footer.rs

```diff
@@ -83,6 +93,19 @@ impl FooterWriter {
             .build()
             .expect("Required fields are not set");
 
-        serde_json::to_vec(&file_metadata).context(SerializeJsonSnafu)
+        if self.lz4_compressed {
+            let mut buf = vec![];
```

Comment:
Maybe we can use `serde_json::to_writer` to eliminate a copy since `FrameEncoder` implements `std::io::Write`:
```rs
            let mut buf = vec![];
            serde_json::to_writer(lz4_flex::frame::FrameEncoder::new(&mut buf), &file_metadata);
```